exporter/exporterhelper/metricshelper.go

```diff
@@ -88,48 +89,86 @@ func NumTimeSeries(md consumerdata.MetricsData) int {
 // the number of dropped metrics.
 type PushMetricsData func(ctx context.Context, md pdata.Metrics) (droppedTimeSeries int, err error)
 
+type metricsRequest struct {
+	baseRequest
+	md     pdata.Metrics
+	pusher PushMetricsData
+}
+
+func newMetricsRequest(ctx context.Context, md pdata.Metrics, pusher PushMetricsData) request {
+	return &metricsRequest{
+		baseRequest: baseRequest{ctx: ctx},
+		md:          md,
+		pusher:      pusher,
+	}
+}
+
+func (req *metricsRequest) onPartialError(consumererror.PartialError) request {
+	// TODO: implement this.
+	return req
+}
+
+func (req *metricsRequest) export(ctx context.Context) (int, error) {
+	return req.pusher(ctx, req.md)
+}
+
+func (req *metricsRequest) count() int {
+	_, numPoints := pdatautil.MetricAndDataPointCount(req.md)
+	return numPoints
+}
+
 type metricsExporter struct {
-	baseExporter
-	pushMetricsData PushMetricsData
+	*baseExporter
+	pusher PushMetricsData
 }
 
-func (me *metricsExporter) ConsumeMetrics(ctx context.Context, md pdata.Metrics) error {
-	exporterCtx := obsreport.ExporterContext(ctx, me.exporterFullName)
-	_, err := me.pushMetricsData(exporterCtx, md)
+func (mexp *metricsExporter) ConsumeMetrics(ctx context.Context, md pdata.Metrics) error {
+	exporterCtx := obsreport.ExporterContext(ctx, mexp.cfg.Name())
+	req := newMetricsRequest(exporterCtx, md, mexp.pusher)
+	_, err := mexp.sender.send(req)
 	return err
 }
 
 // NewMetricsExporter creates an MetricsExporter that can record metrics and can wrap every request with a Span.
-// If no internalOptions are passed it just adds the exporter format as a tag in the Context.
-// TODO: Add support for retries.
-func NewMetricsExporter(config configmodels.Exporter, pushMetricsData PushMetricsData, options ...ExporterOption) (component.MetricsExporter, error) {
-	if config == nil {
+// If no internalOptions are passed it just adds the nextSender format as a tag in the Context.
+func NewMetricsExporter(cfg configmodels.Exporter, pushMetricsData PushMetricsData, options ...ExporterOption) (component.MetricsExporter, error) {
+	if cfg == nil {
 		return nil, errNilConfig
 	}
 
 	if pushMetricsData == nil {
 		return nil, errNilPushMetricsData
 	}
 
-	pushMetricsData = pushMetricsWithObservability(pushMetricsData, config.Name())
+	be := newBaseExporter(cfg, options...)
+
+	// Record metrics on the consumer.
+	be.qSender.nextSender = &metricsSenderWithObservability{
```

Comment:
Same comment as before: is this sender intended to know about the which specific sender it should wrap? Shouldn't it wrap `be.sender`?

Reply:
Yes, as explained previously. For observability metrics/traces we really need to know. That is unfortunate but I don't see other way.